File: solution/java/day23/src/test/java/org/craftedsw/tripservicekata/user/UserBuilder.java
package org.craftedsw.tripservicekata.user;

import io.vavr.collection.Seq;
import org.craftedsw.tripservicekata.trip.Trip;

import static io.vavr.collection.List.empty;
import static io.vavr.collection.List.of;

public class UserBuilder {
    private Seq<User> friends = empty();
    private Seq<Trip> trips = empty();

    public static UserBuilder aUser() {
        return new UserBuilder();
    }

    public UserBuilder friendsWith(User... friends) {
        this.friends = of(friends);
        return this;
    }

    public UserBuilder travelledTo(Trip... trips) {
        this.trips = of(trips);
        return this;
    }

    public User build() {
        return new User(trips, friends);
    }
}

File: solution/c#/Day23/Day23.Tests/Users/UserBuilder.cs
using Day23.Trips;
using Day23.Users;
using LanguageExt;
using static LanguageExt.Seq;

namespace Day23.Tests.Users
{
    public class UserBuilder
    {
        private Seq<User> _friends = empty<User>();
        private Seq<Trip> _trips = empty<Trip>();

        public static UserBuilder AUser() => new();

        public UserBuilder FriendsWith(params User[] friends)
            => this.Let(builder => builder._friends = friends.ToSeq());

        public UserBuilder TravelledTo(params Trip[] trips)
            => this.Let(builder => builder._trips = trips.ToSeq());

        public User Build() => new User(_trips, _friends);
    }
}